### PR TITLE
bugfix: batch_size parameter for DataModules remaining

### LINKED ISSUE #334: Several datamodules ignoring batch_size
## 🐛 Bug

Realized `MNISTDataModule` was ignoring `batch_size` parameter. I found a closed issue (#171) referring to that without a fix.

While fixing it myself (PR #331), I found more datamodules had this problem too - `MNISTDataModule`, `BinaryMNISTDataModule`, `FashionMNISTDataModule`, `SklearnDataModule`, `SSLImagenetDataModule`.

I could take care of that. My question is, is this signature from MNIST in use anymore?:
```python
def train_dataloader(self, batch_size=64, transforms=None):
```
That is basically where the bug comes from. Other datamodules working fine, like `CIFAR10DataModule`, are simply:
```python
def train_dataloader(self):
```

Same thing goes for `val_dataloader` and `test_dataloader`.

### To Reproduce

1. Set up `MNISTDataModule` to use a batch size anything else than 32.
2. Run an experiment
3. Check the batch sizes are 32 regardless

#### Code sample

```python
from pytorch_lightning.core.lightning import LightningModule
from pytorch_lightning import Trainer
from pl_bolts.datamodules.mnist_datamodule import MNISTDataModule

batch_size = 64


class SampleModel(LightningModule):
    def configure_optimizers(self):
        pass

    def training_step(self, batch, batch_idx):
        x, y = batch
        assert x.shape[0] == batch_size
        print('fine')


model = SampleModel()
dm = MNISTDataModule(data_dir='~/Datasets/', batch_size=batch_size)

trainer = Trainer()
trainer.fit(model, dm)
```

### Expected behavior
Batch sizes of 64.
Printing `fine` instead of assert error.

### Environment

 - PyTorch Version (e.g., 1.0): 1.7
 - OS (e.g., Linux): Ubuntu 18.04
 - How you installed PyTorch (`conda`, `pip`, source): conda
 - Build command you used (if compiling from source):
 - Python version: 3.8
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information: pytorch-lightning-bolts 0.2.5

### PR DESCRIPTION
**Pull request by hecoding**
## What does this PR do?

Fixes #334
This PR fixes data modules the same way as PR #331. On issue #334 a number of modules are listed apart from MNIST one, so all them are now fixed and their docstrings updated.

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
In principle it looks fine, just fix falling tests:
```
FAILED pl_bolts/datamodules/sklearn_datamodule.py::pl_bolts.datamodules.sklearn_datamodule.SklearnDataModule
FAILED tests/models/test_executable_scripts.py::test_cli_run_lin_regression[--max_epochs 1 --max_steps 2]
FAILED tests/models/test_executable_scripts.py::test_cli_run_log_regression[--max_epochs 1 --max_steps 2]
```

**Comment by hecoding** [comment]
Oops my bad. I think everything's fine now.

File: pl_bolts/datamodules/binary_mnist_datamodule.py [hunk 1]
```diff
@@ -110,23 +109,19 @@ def train_dataloader(self, batch_size=32, transforms=None):
...
+            batch_size=self.batch_size,
             shuffle=True,
             num_workers=self.num_workers,
             drop_last=True,
             pin_memory=True
         )
         return loader
 
-    def val_dataloader(self, batch_size=32, transforms=None):
+    def val_dataloader(self,):
```
Comment (akihironitta) [inline pl_bolts/datamodules/binary_mnist_datamodule.py:120]:
```suggestion
    def val_dataloader(self):
```

File: pl_bolts/datamodules/binary_mnist_datamodule.py [hunk 2]
```diff
@@ -136,36 +131,32 @@ def val_dataloader(self, batch_size=32, transforms=None):
...
+            batch_size=self.batch_size,
             shuffle=False,
             num_workers=self.num_workers,
             drop_last=True,
             pin_memory=True
         )
         return loader
 
-    def _default_transforms(self):
+    def default_transforms(self):
```
Comment (akihironitta) [inline pl_bolts/datamodules/binary_mnist_datamodule.py:159]: Not sure if we should make this method public...

**Review by akihironitta — COMMENTED** [review]
Thank you for your contribution!

File: pl_bolts/datamodules/binary_mnist_datamodule.py (hunk 2 above)
Comment (hecoding) [reply pl_bolts/datamodules/binary_mnist_datamodule.py:159]: I updated the method to match with the rest of the data modules... Maybe we could get some feedback about the scope from somebody else. `LightningDataModule` abstract class has no actual definition for it so :shrug: 

cc @Borda

Comment (annikabrundyn) [reply pl_bolts/datamodules/binary_mnist_datamodule.py:159]: I think this should just be internal

**Comment by annikabrundyn** [comment]
Aside from keeping the default_transforms as internal, everything else looks good

File: pl_bolts/datamodules/binary_mnist_datamodule.py (hunk 2 above)
Comment (hecoding) [reply pl_bolts/datamodules/binary_mnist_datamodule.py]: gotcha

**Comment by hecoding** [comment]
Ok fixed that thanks, I think that's it.

**Comment by hecoding** [comment]
ping @annikabrundyn

**Comment by annikabrundyn** [comment]
looks great! thanks for the contribution @hecoding :)

**Review by annikabrundyn — APPROVED** [review]

**Comment by hecoding** [comment]
Awesome! Welcome guys.

cc @Borda for merging

**Review by akihironitta — APPROVED** [review]

**Comment by akihironitta** [comment]
@hecoding Thank you :]

**Pull request merged**